### PR TITLE
fix(curriculum): update challenge to heading tags instead of headers

### LINKED ISSUE #43427: Exercise about heading tags <h1> ... <h6>, accidentally refers to header tags <header>
## Affected page

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md

https://www.freecodecamp.org/learn/responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag

## Expected behavior

In the title and the first paragraph, header tags `<header>` are mentioned, but the exercise is about heading tags `<h1> ... <h6>`.

### PR DESCRIPTION
**Pull request by Anurag5086**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #43427

### CODE REVIEW

**Comment by Sembauke** [comment]
Hey @Anurag5086,

I mentioned you in the linked issue, as for now, I will put the blocked label on it.

**Comment by Anurag5086** [comment]
> Hey @Anurag5086,
> 
> I mentioned you in the linked issue, as for now, I will put the blocked label on it.

Sure! Sounds good!

**Review by raisedadead — DISMISSED** [review]
Blocked pending conclusive decision on the original issue.

File: curriculum/challenges/_meta/applied-visual-design/meta.json
Comment (ieahleen) [inline curriculum/challenges/_meta/applied-visual-design/meta.json:50]:
```suggestion
      "Adjust the Size of a Heading Element Versus a Paragraph Element"
```

**Comment by ieahleen** [comment]
You will want to read about how to change the title of a challenge:
https://contribute.freecodecamp.org/#/curriculum-file-structure

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md:26]:
```suggestion
Camper Cat wants a page on his site dedicated to becoming a ninja. Help him fix the headings so his markup gives semantic meaning to the content, and shows the proper parent-child relationships of his sections. Change all the `h5` tags to the proper heading level to indicate they are subsections of the `h2` ones. Use `h3` tags for the purpose.
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md:30]:
```suggestion
Your code should have 6 `h3` elements.
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md:36]:
```suggestion
Your code should have 6 `h3` closing tags.
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md:42]: As this test is not testing for tags (it is testing for elements), the test text should represent this.

```suggestion
Your code should not have any `h5` elements.
```

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.md:48]:
```suggestion
Your code should not have any `h5` closing tags.
```

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md:3]: We should not rename the title without renaming the file.

Also, this cannot be merged, until we have the redirects ready for backlinks.

File: curriculum/challenges/english/01-responsive-web-design/basic-html-cat-photo-app/part-002.md
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/basic-html-cat-photo-app/part-002.md:54]:
```suggestion
Your `h2` element's text should be 'Cat Photos'. Only place the text `Cat Photos` between the opening and closing `h2` tags.
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Thank you, for taking the time to contribute. Please see the requested changes:
- Where there are suggestions, you can click the _Commit suggestion_ button to automatically include the commit
- When a PR has multiple suggestions, you can batch all of the suggestions together, when you are in the `files` view

Lastly, to reiterate one of the suggestions, we should not rename a title without renaming the file, and we will need redirects in place for backlinks, in order to push this to production.

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md
Comment (gikf) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md:3]:
```suggestion
title: Adjust the Size of a Heading Element Versus a Paragraph Element
```
Per suggestion from meta file https://github.com/freeCodeCamp/freeCodeCamp/pull/43429/files#r706828263

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md
Comment (gikf) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md:7]:
```suggestion
dashedName: adjust-the-size-of-a-heading-element-versus-a-paragraph-element
```

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md
Comment (Anurag5086) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md]: I have renamed the file name as well.

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
please follow the steps here on how to rename a challenge: https://contribute.freecodecamp.org/#/curriculum-file-structure

**Comment by ieahleen** [comment]
Also, please, if you are not doing the whole list of challenges to change but only part of it, say so

**Comment by Anurag5086** [comment]
@ieahleen I have made changes in the English language, what should i do about the other languages?

**Comment by Anurag5086** [comment]
> please follow the steps here on how to rename a challenge: https://contribute.freecodecamp.org/#/curriculum-file-structure

I have followed the steps given here. I hope I did them as expected.

**Comment by ieahleen** [comment]
good for the renaming of the challenges you did until now

will you do also all the other challenges on the list?

**Comment by Anurag5086** [comment]
@ieahleen I have made changes for other challenges as well.

**Comment by Anurag5086** [comment]
@ieahleen I have a doubt in this file - /curriculum/challenges/english/03-front-end-development-libraries/sass/store-data-with-sass-variables.md

Not sure what changes should I make here.

**Comment by Anurag5086** [comment]
Not sure why I'm getting a failed test here.
image

**Comment by gikf** [comment]
> Not sure why I'm getting a failed test here.

That's likely due to changed file name for the English language, but still having previous file name in other languages. Other languages need to have file name changed as well.

**Comment by ieahleen** [comment]
> @ieahleen I have a doubt in this file - /curriculum/challenges/english/03-front-end-development-libraries/sass/store-data-with-sass-variables.md

the seed and solution refer to an `h2` element as an header, it is not used in the tests so it should be trivial to change

**Comment by Anurag5086** [comment]
> > @ieahleen I have a doubt in this file - /curriculum/challenges/english/03-front-end-development-libraries/sass/store-data-with-sass-variables.md
> 
> the seed and solution refer to an `h2` element as an header, it is not used in the tests so it should be trivial to change

Okay, thanks! :)

File: curriculum/challenges/chinese-traditional/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md
Comment (ieahleen) [inline curriculum/challenges/chinese-traditional/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md:7]: file title doesn't match

File: curriculum/challenges/chinese/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md
Comment (ieahleen) [inline curriculum/challenges/chinese/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md:7]: file title doesn't match

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
please follow _all_ the steps on how to change a challenge title, until you do that we can't approve this PR

File: curriculum/challenges/chinese/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md
Comment (Anurag5086) [reply curriculum/challenges/chinese/01-responsive-web-design/applied-visual-design/adjust-the-size-of-a-header-versus-a-paragraph-tag.md:7]: Made changes!

**Comment by Anurag5086** [comment]
@ieahleen Can you guide me through, what needs to be done now?

**Comment by ieahleen** [comment]
wait for it to be reviewed, and then redirects will be need to be set up for the challenge with the changed title

**Comment by Anurag5086** [comment]
> wait for it to be reviewed, and then redirects will be need to be set up for the challenge with the changed title

Sure, thanks! 👍

**Review by majestic-owl448 — DISMISSED** [review]
looks good to me, but it needs the redirect for the challenge with changed name

**Comment by Sembauke** [comment]
Hey @raisedadead,

Can you help @ieahleen and @Anurag5086 on what needs to be done with the redirects?

Thanks

**Comment by raisedadead** [comment]
You would need to add redirects to this file: https://github.com/freeCodeCamp/client-config/blob/master/serve.json

This is not ideal but will do.

**Comment by Sembauke** [comment]
@raisedadead I see,

Though only the titles are changed of the challenges.

**Comment by ieahleen** [comment]
@Sembauke changing the title means changing file name and dashed name, so the url to the challenge changes

**Comment by raisedadead** [comment]
I would like to see what @ojeytonwilliams thinks, but I am currently not keen on adding redirects to the serve config. We could do a one-off Nginx based redirection for broken links.

**Comment by ojeytonwilliams** [comment]
> I would like to see what @ojeytonwilliams thinks, but I am currently not keen on adding redirects to the serve config. We could do a one-off Nginx based redirection for broken links.

Neither solution is great, but I slightly prefer modifying serve.json. That approach means we only need to adjust one file and all languages get the redirects.  If we go the NGINX route we have to duplicate the redirects for each language.

**Comment by raisedadead** [comment]
If you are only concerned about code duplication, NGINX does have ["include"](https://nginx.org/en/docs/ngx_core_module.html#include) statements for modularity. Logically, serve is a worse place, to be honest. I think we should redirect early.

**Comment by ojeytonwilliams** [comment]
@raisedadead we've discussed this a bit before: https://github.com/freeCodeCamp/freeCodeCamp/pull/42596#issuecomment-875579093

The TL;DR was we considered NGINX, routing inside the app and a separate redirection 'layer' (i.e. serve).  The conclusion was that the order of preference was app routing > serve > NGINX, but we tried to get Gatsby to handle the redirects and failed.

Unless you've changed your mind about this, I'm fine with continuing to use serve until either Gatsby changes or we migrate to something else that allows us to handle the redirection in the app.  What do you reckon?

**Comment by raisedadead** [comment]
The context here is this change is going to break backlinks shared outside of "the app", i.e the forum, etc. I would recommend redirecting those early.

**Comment by ojeytonwilliams** [comment]
@raisedadead unless I'm missing something, this is essentially the same problem that we've faced before, so the same solutions should work.

To break it down, I'd recommend the following

- Create a Cypress test to verify that the redirects work.  Something like this: https://github.com/freeCodeCamp/freeCodeCamp/pull/43133/commits/728ae64f2ed56c3f3252f17b7c5c508fdf0f6a43
- Create a PR to client-config to change serve.json. Like this: https://github.com/freeCodeCamp/client-config/pull/5

Once the client-config changes are in, the Cypress tests should pass and we can deploy.

**Comment by raisedadead** [comment]
Sure.

**Comment by scissorsneedfoodtoo** [comment]
Just opened a PR to handle the redirect for the challenge title that's being changed here. I think the only thing we're missing now are tests to verify that the redirect is working.

@Anurag5086, could you add a test similar to this?: https://github.com/freeCodeCamp/freeCodeCamp/commit/728ae64f2ed56c3f3252f17b7c5c508fdf0f6a43

If not, I could go in, add a test, and push it to your PR here.

**Review by scissorsneedfoodtoo — DISMISSED** [review]
Thanks again for all of your hard work on this @Anurag5086. I took a look at your test and other changes, and everything LGTM.

I left a minor suggestion for the description of your test, but it's not necessary at all. Feel free to disregard it.

Also, thank you for placing your test in `cypress/integration/learn/redirects/...` rather than `cypress/integration/legacy/redirects/...` like I did in the `let` and `const` PR. I think this makes more sense, and I'll go back to that PR and move the test to the same directory you're using here.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Everything LGTM :+1: :+1: 

Thanks again @Anurag5086!

**Review by raisedadead — APPROVED** [review]

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍

**Pull request merged**
